src/runtime_src/core/edge/user/aie/graph.cpp

```diff
@@ -79,7 +79,7 @@ graph_type(std::shared_ptr<xrt_core::device> dev, const uuid_t, const std::strin
       rtp.selector_row += 1;
       rtp.ping_row += 1;
       rtp.pong_row += 1;
-      rtps.emplace_back(std::move(rtp));
+      rtps.insert(std::make_pair(rtp.name, std::move(rtp)));
```

Comment:
```suggestion
      rtps.emplace(rtp.name, std::move(rtp));
```

Reply:
Same code fewer key strokes.  But btw is either construct really safe?  Is there any chance rtp.name could have been moved before it is copied, for sure order of evaluation is not defined here. There is enough doubt in my mind that I think a copy of the string should have been done outside insert or emplace, the another std::move on the copied string.

Better, a std::set should have been used with proper compare defined.

Reply:
Done

Reply:
> Same code fewer key strokes. But btw is either construct really safe? Is there any chance rtp.name could have been moved before it is copied, for sure order of evaluation is not defined here. There is enough doubt in my mind that I think a copy of the string should have been done outside insert or emplace, then another std::move on the copied string.

Ouch, you are right!

> Better, a std::set should have been used with proper compare defined.

Brilliant.